materialize-snowflake/snowflake.go

```diff
@@ -568,6 +568,7 @@ func (d *transactor) Acknowledge(ctx context.Context) (*pf.ConnectorState, error
 	var checkpointClear = make(checkpoint)
 	for _, b := range d.bindings {
 		checkpointClear[b.target.StateKey] = nil
+		delete(d.cp, b.target.StateKey)
```

Comment:
With this change, I don't think we need the `if len(item.Query) == 0 {` check above anymore, on line 531.